### PR TITLE
[Feature sets] Statistics table header overflows the table on small screen

### PR DESCRIPTION
**Pull request by vasyl-kruk**
https://trello.com/c/pohh2W99/921-feature-sets-statistics-table-header-overflows-the-table-on-small-screen

### CODE REVIEW

File: src/components/Details/details.scss [hunk 1]
```diff
@@ -99,6 +99,41 @@
         background-color: $white;
         border-bottom: $secondaryBorder;
 
+        .arrow {
+          min-width: fit-content;
+          max-width: 24px;
+          //background-color: $silver;
+          //border-radius: 50%;
```
Comment (iurii-vatazhyshyn) [inline src/components/Details/details.scss]: remove comments

File: src/elements/DetailsMenu/DetailsMenu.js [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+  id,
+  iter,
+  match,
+  name,
+  onClick,
+  page
+}) => {
+  const [activeIndex, setActiveIndex] = useState(0)
+  const [windowSize, setWindowSize] = useState(window.innerWidth)
+  const [hideArrows, setHideArrows] = useState(false)
```
Comment (iurii-vatazhyshyn) [inline src/elements/DetailsMenu/DetailsMenu.js]: rename to isHiddenArrows setHiddenArrows

File: src/elements/DetailsMenu/DetailsMenu.js [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+  const details = React.useRef(null)
+  const menu = React.useRef(null)
```
Comment (iurii-vatazhyshyn) [inline src/elements/DetailsMenu/DetailsMenu.js]: rename to detailsRef and menuRef

File: src/elements/DetailsMenu/DetailsMenu.js [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+      return 0
+    } else if (
+      details.current?.scrollWidth >
+      (-activeIndex / 2 + 1) * details.current?.offsetWidth
+    ) {
+      return activeIndex * details.current?.offsetWidth * 0.5
+    } else {
+      return -details.current?.scrollWidth + details.current?.offsetWidth
+    }
+  }
```
Comment (iurii-vatazhyshyn) [inline src/elements/DetailsMenu/DetailsMenu.js]: take it to useState

File: src/elements/DetailsMenu/DetailsMenu.js [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+        onClick={() => {
+          setActiveIndex(prevState => prevState + 1)
+        }}
+      />
+      <div className="details-menu" ref={details}>
+        {detailsMenu.map(link => (
+          <Link
+            style={{
+              transform: `translateX(${transformItems()}px)`,
+              transition: '0.6s'
```
Comment (iurii-vatazhyshyn) [inline src/elements/DetailsMenu/DetailsMenu.js]: set 0.2s instead 0.6s

File: src/elements/DetailsMenu/DetailsMenu.js [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+}) => {
+  const [arrowsAreHidden, setArrowsAreHidden] = useState(true)
+  const [scrolledWidth, setScrolledWidth] = useState(0)
+  const [rightArrowDisabled, setRightArrowDisabled] = useState(false)
+  const tabsWrapperRef = createRef()
+  const tabsRef = createRef()
+
+  const leftArrowClassNames = classnames(
+    'details-menu__arrow',
+    'details-menu__arrow-left',
```
Comment (mariana-furyk) [inline src/elements/DetailsMenu/DetailsMenu.js]:
```suggestion
    'details-menu__arrow_left',
```

File: src/elements/DetailsMenu/DetailsMenu.js [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+    rightArrowDisabled && 'details-menu__arrow_disabled'
+  )
+
+  const scrollTabs = toRight => {
+    let scrollWidth
+
+    if (toRight) {
+      if (
+        tabsRef.current?.scrollWidth <
+        tabsWrapperRef.current?.offsetWidth * 1.5 + scrolledWidth
```
Comment (mariana-furyk) [inline src/elements/DetailsMenu/DetailsMenu.js]: no magic numbers, add numbers to const

File: src/elements/DetailsMenu/DetailsMenu.js [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+          ? 0
+          : scrolledWidth - tabsWrapperRef.current?.offsetWidth / 2
+
+      setRightArrowDisabled(false)
+    }
+
+    setScrolledWidth(scrollWidth)
+  }
+
+  const switchHideArrows = useCallback(() => {
```
Comment (mariana-furyk) [inline src/elements/DetailsMenu/DetailsMenu.js]:
```suggestion
  const handleHideArrows = useCallback(() => {
```

File: src/elements/DetailsMenu/DetailsMenu.js [hunk 9]
```diff
@@ -0,0 +1,200 @@
...
+      )
+    }
+
+    if (scrollIsHidden) {
+      setScrolledWidth(0)
+      setRightArrowDisabled(false)
+    }
+  }, [rightArrowDisabled, tabsRef, tabsWrapperRef])
+
+  const onSelectedTab = useCallback(() => {
```
Comment (mariana-furyk) [inline src/elements/DetailsMenu/DetailsMenu.js]:
```suggestion
  const moveToSelectedTab = useCallback(() => {
```

File: src/elements/DetailsMenu/DetailsMenu.js [hunk 10]
```diff
@@ -0,0 +1,200 @@
...
+    const selectedTab = document.querySelector(
+      `[data-tab='${match.params.tab}']`
+    )
+    const centeredTab =
```
Comment (mariana-furyk) [inline src/elements/DetailsMenu/DetailsMenu.js]:
```suggestion
    const centeredTabPosition =
```

File: src/elements/DetailsMenu/DetailsMenu.js [hunk 11]
```diff
@@ -0,0 +1,200 @@
...
+    window.addEventListener('resize', switchHideArrows)
+
+    return () => window.removeEventListener('resize', switchHideArrows)
+  }, [switchHideArrows])
+
+  useEffect(() => {
+    switchHideArrows()
+  }, [detailsMenu, switchHideArrows])
+
+  useEffect(() => {
```
Comment (mariana-furyk) [inline src/elements/DetailsMenu/DetailsMenu.js:131]: switch places with the above useEffect

**Pull request closed without merging**